Modules/SaltStackConfig/SaltStackConfig.psm1

```diff
@@ -0,0 +1,304 @@
+  .SYNOPSIS
+    This wrapper function will call Get-SscData minions.get_minion_cache.
+  .DESCRIPTION
+    This wrapper function will return the grain property cache of SaltStack Config minions.
+  .EXAMPLE
+    PS C:\> Get-SscMinion
+#>
+  param(
+    [ValidateSet('RAW','Results')][string]$Return='RAW'
+  )
+
+  $output = Get-SscData minions get_minion_cache
+
+  if ($return -eq 'Results') {
+    $output.ret.results
+  } else {
+    $output
+  } # end if for results parameter
+}
+
+Function Get-SscJob {
+<#
+  .NOTES
+  ===========================================================================
+   Created by:	Brian Wuchner
+   Date:		November 27, 2021
+   Blog:		www.enterpriseadmins.org
+   Twitter:		@bwuch
+  ===========================================================================
+  .SYNOPSIS
+    This wrapper function will call Get-SscData job.get_jobs.
+  .DESCRIPTION
+    This wrapper function will return configured SatlStack Config jobs.
+  .EXAMPLE
+    PS C:\> Get-SscJob
+#>
+  param(
+    [ValidateSet('RAW','Results')][string]$Return='RAW'
+  )
+
+  $output = Get-SscData job get_jobs
+
+  if ($return -eq 'Results') {
+    $output.ret.results
+  } else {
+    $output
+  } # end if for results parameter
+}
+
+Function Get-SscSchedule {
+<#
+  .NOTES
+  ===========================================================================
+   Created by:	Brian Wuchner
+   Date:		November 27, 2021
+   Blog:		www.enterpriseadmins.org
+   Twitter:		@bwuch
+  ===========================================================================
+  .SYNOPSIS
+    This wrapper function will call Get-SscData schedule.get.
+  .DESCRIPTION
+    This wrapper function will return schedules for SaltStack Config.
+  .EXAMPLE
+    PS C:\> Get-SscSchedule
+#>
+  param(
+    [ValidateSet('RAW','Results')][string]$Return='RAW'
+  )
+
+  $output = Get-SscData schedule get
+
+  if ($return -eq 'Results') {
+    $output.ret.results
+  } else {
+    $output
+  } # end if for results parameter
+}
+
+Function Get-SscReturn {
+<#
+  .NOTES
+  ===========================================================================
+   Created by:	Brian Wuchner
+   Date:		November 27, 2021
+   Blog:		www.enterpriseadmins.org
+   Twitter:		@bwuch
+  ===========================================================================
+  .SYNOPSIS
+    This wrapper function will call Get-SscData ret.get_returns with either Jid or MinionID.
+  .DESCRIPTION
+    This wrapper function will return job results from the job cache based on the provided arguments.
+  .EXAMPLE
+    PS C:\> Get-SscReturn -Jid '20211122160147314949'
+    PS C:\> Get-SscReturn -MinionID 't147-win22-01.lab.enterpriseadmins.org'
+#>
+  param(
+    [ValidateSet('RAW','Results')][string]$Return='RAW',
+  [string]$jid,
+  [string]$minionid
+  )
+  # ToDo: This should be a parameterset, was having trouble with making the parameters optional.  Use if statement for now

```

Comment:
Perhaps this should be clarified with an additional example.  When `Get-SscReturn` is called without argument, a longer list is returned.  Passing in jid or minionid parameters are both optional and limit the results returned.  I'll add an example without argument, but let me know if the parameter set is still a better option and I'll try to incorporate it.

Reply:
If it doesn't make sense to pass both parameters at the same time having them in separate parameter sets makes sense. What I'm saying is that you cannot make them both optional from the PowerShell perspective, but effectively they will both be optional. You'll have one parameter set where jid will be mandatory parameter, so if it's passed PowerShell will switch to this parameter set. You'll also have another one where minionid will be optional, so if it's passed PowerShell will switch to that parameter set. If none of them is passed PowerShell will determine parameter set to be the one of minionid, but since it's optional it doesn't need to be passed, so the same parameter set will cover both cases - when minionid is passed and when nothing is passed. If both of them are made optional in their parameter sets PowerShell cannot determine the parameter set when no parameter is passed and fails.

Reply:
Thanks for the explanation on how to make this work -- I implemented it as you described and it worked exactly like I had wanted the function to behave.  Thanks so much for the tip.  
In the end though, I've actually backed out this restriction.  In testing, I found that providing both a jid and minion_id does work, and further filters the results.  This is a much better experience and I wouldn't have got here without your help.  This will be resolved in the next commit as well.